active/0021-transfer-files-over-mqtt.md

```diff
@@ -0,0 +1,175 @@
+# MQTT based file transfer
+
+## Changelog
+
+* 2022-09-20: @qzhuyan, @id Initial draft
+
+## Abstract
+
+TODO:
+
+## Motivation
+
+EMQX customers are asking for file transfer functionality from IoT devices to their cloud (primary use case), and from cloud to IoT devices over MQTT. Right now they are uploading files from devices via FTP or HTTPS (e.g. to S3), but it's not working well. For example, both FTP and HTTP servers usually struggle to keep with large number of simultaneous bandwith-intensive connections. Another issue is that uploading files over weak/spotty/unreliable network is not a good experience due to frequent connection reset and transfer restarts.
```

Comment:
More reasons (better to transform the reasons into bullets)
- Adding features to use cases where MQTT is already in place, if other protocols are to be used, it's usually a very different tech stack to deal with. 
  - Another set of programming SDK
  - Different types of security aspects including authentication and authorisation